src/main/java/frc/robot/Logger.java

```diff
@@ -0,0 +1,145 @@
+package frc.robot;
+
+import java.util.Arrays;
+import java.util.List;
+
+import edu.wpi.first.wpilibj.DriverStation;
+
+/**
+ * The Logger class lets you log messages and only have them show up if a config
+ * file entry enables it. It supports the following logging levels: - ERROR -
+ * WARN - INFO (the default) - VERBOSE - DEBUG - SILLY You can select the minimum level at
+ * which to allow logging. For example, if log.myLogger is INFO, then only logs
+ * with the levels of WARN, ERROR, or INFO will show up. Set the log level in
+ * the config to OFF (or don't set it at all) to disable logging for this logger
+ * entirely. See Config.java for more info about configuration.
+ * 
+ * @example Logger myLogger = new Logger("myLogger");
+ * 
+ *          // if log.myLogger is INFO or lower, this prints to STDOUT
+ *          [myLogger:INFO] Hello from myLogger! myLogger.log("Hello from
+ *          myLogger!");
+ * 
+ *          // prints to STDERR [myLogger:ERR] Error from myLogger!
+ *          myLogger.err("Error from myLogger!");
+ */
+public class Logger {
```

Comment:
The point of this class is only to log to the driver station? Is there any value to logging to a file also for analysis? Does the DriverStation automatically log to a file on the driverstation machine?

Reply:
It only logs to the driver station, which stores logs in files on the driver station machine.